### PR TITLE
Remove custom BoringSSL build

### PR DESCRIPTION
**Pull request by reedloden**
The `boring` crate will compile BoringSSL on demand.

Remove unneeded Clang 7 build and replace Clang 10 with Clang 12. BoringSSL in FIPS mode explicitly requires Clang 12.0.0, while libbpf and related tools only require Clang 10+, so standardized everything on Clang 12.0.0 so that we don't need multiple Clang installations.

This also required libbpf to be bumped, as 1.0.1 no longer compiled. Both 1.1.x and 1.2.x seem to build fine, so went ahead and bumped to 1.2.2 (latest libbpf). As a result, `aquasecurity/libbpfgo` was also bumped to match the new version.

Additionally, add a few missing git commit hash validations that were noticed as all the `Dockerfile`s were being reviewed/updated.

### CODE REVIEW

**Review by probakowski — APPROVED** [review]

File: build.assets/Dockerfile-centos7-fips [hunk 1]
```diff
@@ -59,17 +69,17 @@ ENV GOPATH="/go" \
...
 
 RUN mkdir -p /opt && cd /opt && \
     git clone https://github.com/google/boringssl.git && \
     cd boringssl && \
-    git checkout ae223d6138807a13006342edfeef32e813246b39 && \
+    git checkout 853ca1ea1168dff08011e5d42d94609cc0ca2e27 && \
     mkdir build && \
     cd build && \
-    scl enable llvm-toolset-7.0 "cd /opt/boringssl/build && cmake3 -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DFIPS=1 -DCMAKE_BUILD_TYPE=Release -GNinja .. && ninja"
+    scl enable llvm-toolset-13.0 "cd /opt/boringssl/build && cmake3 -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DFIPS=1 -DCMAKE_BUILD_TYPE=Release -GNinja .. && ninja"
```
Comment (zmb3) [inline build.assets/Dockerfile-centos7-fips:82]: Why 13? Don't we need clang 12?

**Review by zmb3 — COMMENTED** [review]
This seems incomplete. I would expect a rdp-rs update too in order to pick up the newer boring crate that includes https://github.com/cloudflare/boring/commit/902e7d0c926facd43c17e42012e9ce5b91e66184

File: build.assets/Dockerfile-centos7-fips [hunk 2]
```diff
@@ -42,14 +43,23 @@ RUN yum groupinstall -y 'Development Tools' && \
...
 
+RUN cat << EOF > /etc/yum.repos.d/llvmtoolset-build.repo
+[llvmtoolset-build]
+name=LLVM Toolset 13.0 - Build
+baseurl=https://buildlogs.centos.org/c7-llvm-toolset-13.0.x86_64/
+gpgcheck=0
+enabled=1
+EOF
+
+RUN yum install --nogpgcheck llvm-toolset-13.0-clang-tools-extra llvm-toolset-13.0-clang
```
Comment (zmb3) [inline build.assets/Dockerfile-centos7-fips:56]: Missing the `-y` flag.

File: build.assets/Dockerfile-centos7-fips (hunk 1 above)
Comment (zmb3) [inline build.assets/Dockerfile-centos7-fips:82]: Come to think of it, I'm not sure we need to clone and build boringssl and then set `BORING_BSSL_PATH` at all.

https://docs.rs/boring/latest/boring/#support-for-pre-built-binaries-or-custom-source says:

> While this crate can build BoringSSL on its own, you may want to provide pre-built binaries instead. To do so, specify the environment variable BORING_BSSL_PATH with the path to the binaries.
>
> You can also provide specific headers by setting BORING_BSSL_INCLUDE_PATH.
>
> Notes: The crate will look for headers in the $BORING_BSSL_INCLUDE_PATH/openssl/ folder, make sure to place your headers there.
>
> In alternative a different path for the BoringSSL source code directory can be specified by setting BORING_BSSL_SOURCE_PATH which will automatically be compiled during the build process.

And then later on it says:

> Only BoringCrypto module version 853ca1ea1168dff08011e5d42d94609cc0ca2e27, as certified with [FIPS 140-2 certificate 4407](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4407) is supported by this crate. Support is enabled by this crate’s fips feature.

So it sounds to me like we just need to update the boring dependency, ensure that the fips feature is enabled (it is), and stop pre-building.

Do you interpret that the same?

Comment (zmb3) [reply build.assets/Dockerfile-centos7-fips:82]: Still waiting your response on this one.

Comment (reedloden) [reply build.assets/Dockerfile-centos7-fips]: Sorry, I fixed this locally (which is why I resolved this), but just didn't push my change back.

File: build.assets/Dockerfile-centos7-fips [hunk 3]
```diff
@@ -96,8 +119,6 @@ RUN yum groupinstall -y 'Development Tools' && \
     elfutils-libelf-devel-static \
     git \
     net-tools \
-    # required to create bindings for Rust's boring-rs crate
```
Comment (reedloden) [inline build.assets/Dockerfile-centos7-fips:99]: Seems like we're including `/opt/llvm` via line 187 anyway, so this doesn't seem needed, as we overwrite it.

File: build.assets/Dockerfile-centos7-fips (hunk 1 above)
Comment (zmb3) [reply build.assets/Dockerfile-centos7-fips]: Pinning Go was an artifact of Go pre-1.19 where you had to use an old version of Go to build/bootstrap a new version of Go that's FIPS compliant. 

Now that there's no longer a separate FIPS Go toolchain I think that's probably unnecessary. 

Plus, this Rust code has nothing to do with Go. 

+1 for relying on boring to handle this all.

Comment (reedloden) [reply build.assets/Dockerfile-centos7-fips]: Sounds good. I was able to clean-up a fair bit of stuff. libbpf and related tools only require Clang 10+, so standardized everything on Clang 12.0.0 so that we don't need multiple Clang installations.

**Comment by reedloden** [comment]
> This seems incomplete. I would expect a rdp-rs update too in order to pick up the newer boring crate that includes [cloudflare/boring@902e7d0](https://github.com/cloudflare/boring/commit/902e7d0c926facd43c17e42012e9ce5b91e66184)

This is being handled in https://github.com/gravitational/rdp-rs/pull/35. There's also https://github.com/gravitational/teleport/pull/33900 as well.

File: build.assets/Dockerfile [hunk 4]
```diff
@@ -251,9 +251,11 @@ RUN make -C /opt/pam_teleport install
 ENV SOFTHSM2_PATH "/usr/lib/softhsm/libsofthsm2.so"
 
 # Install bats.
-RUN curl -fsSL https://github.com/bats-core/bats-core/archive/v1.2.1.tar.gz | tar -xz && \
-    cd bats-core-1.2.1 && ./install.sh /usr/local && cd .. && \
-    rm -r bats-core-1.2.1
+RUN git clone --depth=1 https://github.com/bats-core/bats-core.git -b v1.2.1 && \
```
Comment (zmb3) [inline build.assets/Dockerfile:254]: Why the change from downloading a tarball to git cloning the source?

Don't you think these changes belong in a separate PR?I don't see how they relate to removing the custom BoringSSL build.

**Review by zmb3 — APPROVED** [review]
Good cleanup.

I do wonder if we should put a comment anywhere we reference a specific version that's required by the FIPS certificate so that people don't end up changing the versions in the future without understanding why we're pinned to something older.

**Comment by zmb3** [comment]
Also, I would run a tag build before merging this to make sure it plays well with the version of rdp-rs used on master.

File: build.assets/Dockerfile (hunk 4 above)
Comment (reedloden) [reply build.assets/Dockerfile:254]: I was reviewing all the `Dockerfile`s to make sure I wasn't missing anything, and I noticed this discrepancy. I can split it out, but was kinda hoping to keep it as a /boyscout/ fix.

**Comment by reedloden** [comment]
> I do wonder if we should put a comment anywhere we reference a specific version that's required by the FIPS certificate so that people don't end up changing the versions in the future without understanding why we're pinned to something older.

Added.

**Comment by reedloden** [comment]
@jakule Going from Clang 10 to Clang 12 caused libbpf 1.0.1 to not compile. I bumped it to 1.1.x, which works fine, but I also tried 1.2.x as well, and that seems to work (solely from build and existing tests). Do you have a preference on which libbpf version to use?

File: build.assets/Makefile [hunk 5]
```diff
@@ -71,13 +71,6 @@ endif
 DOCKERFLAGS := $(DOCKERFLAGS) -v $(GOCACHE):/go/cache -e GOCACHE=/go/cache
 endif
 
-ifeq ("$(RUNTIME_ARCH)","arm64")
-	# devtoolset-11 is not available on CentOS 7 aarch64
-	DEVTOOLSET ?= devtoolset-10
-else
-	DEVTOOLSET ?= devtoolset-11
-endif
```
Comment (jakule) [inline build.assets/Makefile:79]: Why was this removed? AFAIR it was needed for aarch64 build to work.

Comment (reedloden) [reply build.assets/Makefile:79]: `devtoolset-12` has aarch64, so this is no longer needed.

**Review by jakule — APPROVED** [review]
